samples/subsys/usb/cdc_acm/src/main.c

```diff
@@ -106,12 +108,19 @@ static void interrupt_handler(const struct device *dev, void *user_data)
 	ARG_UNUSED(user_data);
 
 	while (uart_irq_update(dev) && uart_irq_is_pending(dev)) {
-		if (uart_irq_rx_ready(dev)) {
+		if (!rx_throttled && uart_irq_rx_ready(dev)) {
 			int recv_len, rb_len;
 			uint8_t buffer[64];
 			size_t len = MIN(ring_buf_space_get(&ringbuf),
 					 sizeof(buffer));
 
+			if (len == 0) {
+				/* Throttle because ring buffer is full */
+				uart_irq_rx_disable(dev);
+				rx_throttled = true;
+				continue;
+			}
```

Comment:
Wouldn't `} else {` be more appropriate here so that the TX path could be handled without prior calls to `uart_irq_update` and `uart_irq_is_pending`?

Reply:
Not quite sure what you mean, but uart_irq_update()  must be called before uart_irq_tx_ready().

Reply:
I don't think such micro-optimization would make any sense.